src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2974,7 +2980,7 @@ private BoundStatement BindSwitchStatement(SwitchStatementSyntax node, Diagnosti
         {
             Debug.Assert(node != null);
             Binder switchBinder = this.GetBinder(node);
```

Comment:
In release, compiler would crash anyway. Asserts in the new code make it easier to debug issues with it, a dialog pops-up with an option to attach a debugger.